router/registry/registry.go

```diff
@@ -62,17 +64,17 @@ func NewRouter(opts ...router.Option) router.Router {
 
 // Init initializes router with given options
 func (r *rtr) Init(opts ...router.Option) error {
-	// stop the router before we initialize
-	if err := r.Close(); err != nil {
-		return err
-	}
-
 	r.Lock()
-	defer r.Unlock()
-
 	for _, o := range opts {
 		o(&r.options)
```

Comment:
Should we reset options on Init so that we don't have any hangover from previous inits?